src/arena.c

```diff
@@ -1259,13 +1258,16 @@ static size_t
 arena_prof_demote(tsdn_t *tsdn, edata_t *edata, const void *ptr) {
 	cassert(config_prof);
 	assert(ptr != NULL);
+	size_t usize = isalloc(tsdn, ptr);
+	size_t bumped_usize = sz_sa2u(usize, PAGE);
+	szind_t szind = sz_size2index(bumped_usize);
 
-	edata_szind_set(edata, SC_NBINS);
-	emap_remap(tsdn, &arena_emap_global, edata, SC_NBINS, /* slab */ false);
+	edata_szind_set(edata, szind);
+	emap_remap(tsdn, &arena_emap_global, edata, szind, /* slab */ true);
```

Comment:
We should still do `slab = false`, right?

Reply:
Now that you mention it, I don't think either is correct. What if `slab_sizes` was customized in the `MALLOC_CONF`? Say that you have `PAGE` set to 32 KiB, and `MALLOC_CONF=slab_sizes:32768-32768:2`. Setting `slab` to `false` is wrong, since all large size classes are >= 4x `PAGE`. Setting `slab` to `true` in the way that I'm currently doing it would _also_ be wrong, since the `MALLOC_CONF` specifies 2-`PAGE` slabs for the 32 KiB size-class.

Reply:
These edatas are indeed odd; their size is the small class, but not slabs. Setting it to non slab probably is "more" correct -- really the key here is, does it go through the arena bins or any slab "interior" management. In a way small vs large, is more about slab vs non-slab. As you have noticed, the large_alloc handles the <4x PAGE size fine, as long as it's a PAGE size. Setting it a non-slab to make sure it goes back to the large_* functions probably works fine, while setting to slab, as you mentioned, will create a different sized slab that will cause conflicts.